share/lib/python/neuron/rxd/geometry3d/FullJoinMorph.py

```diff
@@ -54,24 +54,36 @@ def fullmorph(source, dx, soma_step=100, mesh_grid=None, relevant_pts=None):
 
     # grid setup
     xs, ys, zs, diams, arcs = [],[],[],[],[]
+    soma_idx = None
     for i, sec in enumerate(source):
         if relevant_pts:
             rng = relevant_pts[i]
         else:
             rng = range(sec.n3d())
-        
+        if 'soma[0]' in sec.hname() and 'soma' in _sec_db:
+            soma_idx = i
+        else:
+            xs += [sec.x3d(i) for i in rng]
+            ys += [sec.y3d(i) for i in rng]
+            zs += [sec.z3d(i) for i in rng]
+            diams += [sec.diam3d(i) for i in rng]
+            arcs += [sec.arc3d(i + 1) - sec.arc3d(i) for i in rng[:-1]]

```

Comment:
Honestly what we need for these things is the minimum and maximum values; there's no need to keep a list of any of them. (And for arcs and diams, we only need the minimum)